101-rds-deployment-full-ha/azuredeploy.json

```diff
@@ -0,0 +1,1180 @@
+                                        }
+                                    }
+                                ]
+                            },
+                            "name": "[concat(parameters('vmName'),copyIndex(1),'nic1')]",
+                            "copy": {
+                                "name": "[concat(parameters('vmName'),'niccopy')]",
+                                "count": "[parameters('loopCount')]"
+                            }                                                        
+                        },
+                        {
+                            "type": "Microsoft.Compute/virtualMachines",
+                            "apiVersion": "2019-07-01",
+                            "location": "[parameters('location')]",
+                            "dependsOn": [
+                                "[concat(parameters('vmName'),copyIndex(1),'nic1')]"
+                            ],
+                            "properties": {
+                                "licenseType": "Windows_Server",
+                                "billingProfile": {
+                                    "maxPrice": -1
+                                },
+                                "diagnosticsProfile": {
+                                    "bootDiagnostics": {
+                                        "enabled": true,
+                                        "storageUri": "[parameters('storageDiagUrl')]"
+                                    }
+                                },
+                                "networkProfile": {
+                                    "networkInterfaces": [
+                                        {
+                                            "id": "[resourceId('Microsoft.Network/networkInterfaces',concat(parameters('vmName'),copyIndex(1),'nic1'))]"
+                                        }
+                                    ]
+                                },
+                                "osProfile": {
+                                    "adminUsername": "[parameters('adminUser')]",
+                                    "adminPassword": "[parameters('adminPasswd')]",
+                                    "computerName": "[concat(parameters('vmName'),copyIndex(1))]"
+                                },
+                                "hardwareProfile": {
+                                    "vmSize": "[parameters('vmSize')]"
+                                },
+                                "priority": "Spot",
+                                "evictionPolicy": "Deallocate",
+                                "storageProfile": {
+                                    "osDisk": {
+                                        "createOption": "FromImage",
+                                        "managedDisk": {
+                                            "storageAccountType": "[parameters('storageSkuType')]"

```

Comment:
you need to separate the parameter for storageAccountSku into two (or just use a variable for the diags storage).  diags only allows for Standard_LRS, whereas vm disks can use premium or ssd disk.